clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1724,16 +1724,21 @@ bool isAValidPVConstraint(const ConstraintVariable *C) {
 
 // Replace CVars and ArgumentConstraints with those in [FromCV].
 void PointerVariableConstraint::brainTransplant(ConstraintVariable *FromCV,
-                                                ProgramInfo &I) {
+                                                ProgramInfo &I,
+                                                std::string &ReasonFailed) {
   PVConstraint *From = dyn_cast<PVConstraint>(FromCV);
   assert(From != nullptr);
   CAtoms CFrom = From->getCvars();
-  assert(Vars.size() == CFrom.size());
+  if(Vars.size() != CFrom.size()) {
+    ReasonFailed = "transplanting different pointer levels";
```

Comment:
This might be better phrased as "transplanting between pointer with different depth"

Reply:
I'll change it